src/hotspot/share/classfile/dictionary.cpp

```diff
@@ -420,7 +417,8 @@ void Dictionary::validate_protection_domain(unsigned int name_hash,
 
 // During class loading we may have cached a protection domain that has
 // since been unreferenced, so this entry should be cleared.
-void Dictionary::clean_cached_protection_domains() {
+void Dictionary::clean_cached_protection_domains(GrowableArray<ProtectionDomainEntry*>* delete_list) {
+  assert(Thread::current()->is_Java_thread(), "only called by JavaThread");
   assert_locked_or_safepoint(SystemDictionary_lock);
```

Comment:
If the current thread must be a JavaThread then we cannot be at a safepoint and so the second assert can just check the lock is held.

Reply:
sure.